src/Microsoft.AspNetCore.Razor.Language/RazorSourceDocument.cs

```diff
@@ -72,6 +90,15 @@ public virtual string GetChecksumAlgorithm()
             return HashAlgorithmName.SHA1.Name;
         }
 
+        /// <summary>
+        /// Gets the file path in a format that should be used for display.
+        /// </summary>
+        /// <returns>The <see cref="RelativePath"/> if set, or the <see cref="FilePath"/>.</returns>
+        public virtual string GetFilePathForDisplay()
+        {
+            return RelativePath ?? FilePath;
```

Comment:
Hmm, is it worth handling the situation when both relative path and file path are null? I forget what we do in documents that have both of those values null.

Reply:
If that happens you just want get a file path in the display. Seems fine to me